### ISSUE TITLE
Socks inside electron

### ISSUE THREAD

**Issue by arj03** [JoshGlazebrook/socks#33]
Hi

First of all, thanks for a really great library!

I have a questions more than an issue, I'm using socks to create a proxy connection to tor running on my machine, this works great in a normal node application, but if I run the same code inside electron, I start seeing timeouts and it seems to be unable to create a proxy connection to tor. Any suggestions what might be the issue?

Thanks!

**Comment by JoshGlazebrook** [comment]
Not entirely sure. Do you get any kind of errors?

**Comment by arj03** [comment]
Sadly it just doesn't work. It seems like it gets blocked or something. I was just wondering if you had heard about this before, as it seems like it would be a case others would have run into.

Thanks

**Comment by JoshGlazebrook** [comment]
I've not heard any other reports, but I've also never used electron. I'll try and set aside some time this weekend to experiment with it a bit and see if I can find anything going on.

**Comment by arj03** [comment]
Thank you, that is very kind.

**Comment by knoxcard** [comment]
@arj03 - I don't see why Electron would be any different as everyone is stating it's just using Javascript/CSS/Chromium and displaying frameless windows on the desktop.  Can you post some of the SOCKS code you are using?

**Comment by arj03** [comment]
@knoxcard this is on linux.

What I'm trying to do is have a [scuttlebot](https://github.com/ssbc/scuttlebot) embedded inside [patchbay](https://github.com/ssbc/patchbay) connect to a [tor](https://www.torproject.org/) socks proxy on the local machine. Everything works fine if scuttlebot is a seperate process. Its only when included inside patchbay (and thus electron) that nothing happen. I think electron must have some kind of sandbox preventing this but I can't find an option for this.

**Comment by marpme** [comment]
https://github.com/JoshGlazebrook/socks/blob/3243255c8a21fdd01328ee2b556ffa8672f18ddf/src/client/socksclient.ts#L281-L284

### In browsers: 
The SetTimeout call returns a simple positive integer referring to the process handling the 
timed out code.

https://developer.mozilla.org/en-US/docs/Web/API/WindowOrWorkerGlobalScope/setTimeout#Return_value

### NodeJS:

But NodeJS handles them as timers (as a part of child_process) 
https://nodejs.org/api/timers.html#timers_timeout_unref

So my solution would be that we have a check whether if we have a this function available on the returned object and then call it if it's there.

## Error that happend (ElectronJS)

```
Uncaught (in promise) TypeError: setTimeout(...).unref is not a function
    at SocksClient.connect (bundle.js:261769)
    at Promise (bundle.js:261605)
    at new Promise (<anonymous>)
    at Function.createConnection (bundle.js:261603)
    at onlookup (bundle.js:261525)
    at SocksProxyAgent.connect [as callback] (bundle.js:261550)
    at bundle.js:115222
    at new Promise (<anonymous>)
    at SocksProxyAgent.callback (bundle.js:115201)
    at SocksProxyAgent.addRequest (bundle.js:1202)
```

**Comment by JoshGlazebrook** [comment]
This is now fixed, and is published to npm as 2.2.2
